pkg/infra/kubernetes/helm_chart.go

```diff
@@ -16,21 +16,22 @@ import (
 
 type HelmChart struct {
 	Name           string
+	Chart          string
 	ValuesFiles    []string
 	ExecutionUnits []*HelmExecUnit
 	Directory      string
 	Files          []core.File
-	Values         []Value
-	AnnotationKeys []core.AnnotationKey
+	Values         []HelmChartValue
+	ConstructRefs  []core.AnnotationKey
 }
 
 // Provider returns name of the provider the resource is correlated to
 func (chart *HelmChart) Provider() string { return "kubernetes" }
 
-// KlothoResource returns AnnotationKey of the klotho resource the cloud resource is correlated to
-func (chart *HelmChart) KlothoConstructRef() []core.AnnotationKey { return chart.AnnotationKeys }
+// KlothoConstructRef returns a slice containing the ids of any Klotho constructs is correlated to
+func (chart *HelmChart) KlothoConstructRef() []core.AnnotationKey { return chart.ConstructRefs }
 
```

Comment:
does helm chart need to be a core.Resource? I see value in it being one, but it doesnt look like it maps to the factory and wouldnt deploy. Im wondering if we just omit it and all other providers require the transformation. 

This would make the kubernetes plugin an analysis and transformation rather than a providerplugin.

Reply:
+1 — I was having the exact same thought as I was scrolling through the above.

The rule we'd discussed earlier is "it's a resource if it has an ARN". I don't think Helm charts do, right? They're more of a document that affects the k8s NodeGroup (?), similar to how a policy document affects an IAM Role.

Reply:
well i do think helm chart is a resource because its a different provider than aws technically. For the kubernetes world it would be anything that has a manifest is a resource and a helm chart contains many manifests.

My point here is more so, we have the AWSHelmChart which is the chart after weve done all conversions for aws resources and thats what we deploy. So having the base helm chart be a resource sounds a bit far fetched, the question becomes how do we support base helm charts then

Reply:
Right now, our provider is for AWS. We have plans to write a different, k8s-only provider — but that's future work. So while we're in the mode of writing the AWS provider, imo we should really try to stick to "resource ⟺ arn-haver".

Reply:
yeah but then we wouldnt deploy any eks exec units or its corresponding resources

Reply:
so helm chart has to be a resource to some degree, we just have the aws helm chart which is the resource and i dont think the generic one needs to be one

Reply:
We talked about this offline, I'm aligned 👍